server: fix TestReportUsage

tbg opened the pull request:
- Fixes a goroutine leak under darwin (when dialing
tcp://instance-data.ec2.internal:80 as we happen to try we end up with
TCP-level timeouts, and that apparently takes long on OSX unless we
override the default)

- Works around a gossip-related flake.

Fixes #38174.
Fixes #38176.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/38205)
<!-- Reviewable:end -->

dt commented on pkg/server/updates.go:
```diff
@@ -454,7 +455,20 @@ func (s *Server) reportDiagnostics(ctx context.Context) {
-	res, err := http.Post(reportingURL.String(), "application/x-protobuf", bytes.NewReader(b))
+
+	const timeout = 3 * time.Second
+	client := &http.Client{
```
Do we need this one? *this* conn shouldn't fail in weird ways should it?

tbg replied:
We don't need it for the test, but using a client without a timeout also seemed like a bad idea. Let me know if you have concerns about this and I'll follow up.

tbg committed:
server: fix TestReportUsage

- Fixes a goroutine leak under darwin (when dialing
tcp://instance-data.ec2.internal:80 as we happen to try we end up with
TCP-level timeouts, and that apparently takes long on OSX unless we
override the default)

- Works around a gossip-related flake.

Fixes #38174.
Fixes #38176.

Release note: None

tbg commented:
bors r=dt
TFTR!
